### PR TITLE
Community: Prevent null billing config on upload

### PR DESCRIPTION
**Pull request by JAVGan**
This commit changes the `community_push.items._set_push_item_billing_code` function to prevent returning a push item with no billing codes.

The function `_set_push_item_billing_code` is only called when the billing config is required and it doesn't make sense to let a push item miss it when this value is required.

It also improve the billing config matching debugging logs to help analyze what could have happened when such push item doesn't have a billing code match.

Refers to SPSTRAT-408

### CODE REVIEW

File: src/pubtools/_marketplacesvm/tasks/community_push/items.py [hunk 1]
```diff
@@ -53,14 +53,19 @@
     out_name = None
 
     for bc_conf_item in billing_config.values():
-        if is_match(bc_conf_item, os.path.basename(push_item.src), push_item.type):
+        base_src = os.path.basename(push_item.src)
+        log.debug("Attempting to match %s to %s type %s", bc_conf_item, base_src, push_item.type)
```
[the remark this answers is not in the document]
Comment (JAVGan) [reply src/pubtools/_marketplacesvm/tasks/community_push/items.py]: Amended commit is no longer logging the entire `bc_conf_item` object, rather just it's name `bc_conf_name` (str)

File: src/pubtools/_marketplacesvm/tasks/community_push/items.py [hunk 2]
```diff
@@ -52,15 +52,25 @@
     out_codes = []
     out_name = None
 
-    for bc_conf_item in billing_config.values():
-        if is_match(bc_conf_item, os.path.basename(push_item.src), push_item.type):
+    for bc_conf_name, bc_conf_item in billing_config.items():
+        base_src = os.path.basename(push_item.src)
+        log.debug(
+            "Attempting to match billing rule %s to %s type %s",
+            bc_conf_name,
```
[the remark this answers is not in the document]
Comment (JAVGan) [reply src/pubtools/_marketplacesvm/tasks/community_push/items.py:59]: Not a sensitive content: out_codes

File: src/pubtools/_marketplacesvm/tasks/community_push/items.py [hunk 3]
```diff
@@ -53,14 +53,19 @@
     out_name = None
 
     for bc_conf_item in billing_config.values():
-        if is_match(bc_conf_item, os.path.basename(push_item.src), push_item.type):
+        base_src = os.path.basename(push_item.src)
+        log.debug("Attempting to match %s to %s type %s", bc_conf_item, base_src, push_item.type)
+        if is_match(bc_conf_item, base_src, push_item.type):
+            log.debug("Matched code %s for %s", bc_conf_item, push_item.src)
```
[the remark this answers is not in the document]
Comment (JAVGan) [reply src/pubtools/_marketplacesvm/tasks/community_push/items.py]: Amended commit is no longer logging the entire `bc_conf_item` object, rather just it's name `bc_conf_name` (str)

File: src/pubtools/_marketplacesvm/tasks/community_push/items.py [hunk 4]
```diff
@@ -52,15 +52,25 @@
...
+    for bc_conf_name, bc_conf_item in billing_config.items():
+        base_src = os.path.basename(push_item.src)
+        log.debug(
+            "Attempting to match billing rule %s to %s type %s",
+            bc_conf_name,
+            base_src,
+            push_item.type,
+        )
+        if is_match(bc_conf_item, base_src, push_item.type):
+            log.debug("Matched billing rule %s for %s", bc_conf_name, push_item.src)
```
[the remark this answers is not in the document]
Comment (JAVGan) [reply src/pubtools/_marketplacesvm/tasks/community_push/items.py:64]: Not a sensitive content: `bc_conf_name`, `out_codes`, `out_name`

**Review by jajreidy — APPROVED** [review]
LGTM

**Pull request merged**